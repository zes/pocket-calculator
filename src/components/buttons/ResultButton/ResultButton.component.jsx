import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import '../button.css';

export default class ResultButton extends PureComponent {
  onClick = () => {
    const { getResult } = this.props;
    getResult();
  };

  render() {
    const { className } = this.props;
    const buttonClassName = `button ${className}`;
    return (
      <button type="button" className={buttonClassName} onClick={this.onClick}>
        =
      </button>
    );
  }
}

ResultButton.propTypes = {
  className: PropTypes.string.isRequired,
  getResult: PropTypes.func.isRequired,
};
