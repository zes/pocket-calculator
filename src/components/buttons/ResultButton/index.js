export { default } from './ResultButton.container';
