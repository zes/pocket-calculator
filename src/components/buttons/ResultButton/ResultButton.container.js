import { connect } from 'react-redux';
import { getResult } from '../../../redux/actions';

import ResultButton from './ResultButton.component';

const mapDispatchToProps = {
  getResult,
};

export default connect(
  null,
  mapDispatchToProps,
)(ResultButton);
