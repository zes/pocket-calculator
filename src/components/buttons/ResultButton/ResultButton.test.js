import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ResultButton from './ResultButton.component';

Enzyme.configure({ adapter: new Adapter() });

describe('[Component] <ResultButton />', () => {
  const props = {
    className: 'equal',
    getResult: jest.fn(),
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<ResultButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call updateOperation when user click on button', () => {
    const resultButton = shallow(<ResultButton {...props} />);
    resultButton.instance().onClick();
    expect(props.getResult).toHaveBeenCalledWith();
  });
});
