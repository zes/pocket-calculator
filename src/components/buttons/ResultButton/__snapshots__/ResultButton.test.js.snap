// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <ResultButton /> renders a snapshot 1`] = `
<button
  className="button equal"
  onClick={[Function]}
  type="button"
>
  =
</button>
`;
